client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,634 @@
+package zio.raft.client
+
+import zio._
+import zio.stream._
+import zio.raft.protocol._
+import scodec.bits.ByteVector
+import java.time.Instant
+
+/**
+ * Main RaftClient implementation using functional state-based approach.
+ * 
+ * Each connection state (Disconnected, Connecting, Connected) handles messages
+ * differently and transitions to other states based on the protocol.
+ * 
+ * Session management is integrated directly into the state machine.
+ */
+class RaftClient private (
+  stateRef: Ref[ClientState],
+  pendingOperationsRef: Ref[Map[Nonce, SessionOperation]],
+  serverRequestTrackerRef: Ref[ServerRequestTracker],
+  retryManager: RetryManager,
+  zmqTransport: ZmqClientTransport,
+  config: ClientConfig
+) {
+  
+  /**
+   * Connect to the cluster and create a session.
+   */
+  def connect(): Task[SessionId] = 
```

Comment:
let's remove this method.

When start raft client, we would internally call the connect method.

Anyway, this is wrong because this type of code must run within the stream, it cannot run from a different thread